src/cargo/ops/cargo_new.rs

```diff
@@ -795,8 +795,27 @@ fn mk(config: &Config, opts: &MkOptions<'_>) -> CargoResult<()> {
                     .push(bin);
             }
         } else if i.relative_path != "src/lib.rs" {
-            manifest["lib"]["name"] = toml_edit::value(i.target_name.clone());
-            manifest["lib"]["path"] = toml_edit::value(i.relative_path.clone());
+            let mut lib = toml_edit::Table::new();
+            lib["name"] = toml_edit::value(i.target_name.clone());
+            lib["path"] = toml_edit::value(i.relative_path.clone());
+            manifest["lib"] = toml_edit::Item::Table(lib);
+        }
+    }
+
+    let manifest_path = path.join("Cargo.toml");
+    if let Ok(root_manifest_path) = find_root_manifest_for_wd(&manifest_path) {
+        let root_manifest = paths::read(&root_manifest_path)?;
+        let workspace_document = root_manifest.parse::<toml_edit::Document>()?;
+        if let Some(workspace_package_keys) = workspace_document
+            .get("workspace")
+            .and_then(|workspace| workspace.get("package"))
```

Comment:
As rust-lang/cargo#12148 is being merged before this, could you update this to also include `workspace.lints`?

Reply:
Do you mind if I add it to the follow-up PR? I need some time to check out this new feature. Also, it is an unstable new feature, do we have to support inheriting it too?

Reply:
Yes, we can do it in a follow up.  On merging this, we should note it in #12115, both that it needs to happen and when its resolved

It is as minimum a blocker for stabilization of #12115 but it would be preferable for us to not let unstable features fall behind as we do feature and bug fixes.  In this case, there is enough active interest (from me at least) that we'd keep up but in general it cane widen the gulf enough to make it that much harder to stabilize things in the future.  For example, I tested `lints` against `profile-rustflags` and made some tweaks.

Reply:
Sounds good. I will try it in my next PR.